.github/workflows/release-please.yml

```diff
@@ -0,0 +1,46 @@
+on:
+  push:
+    branches:
+      - dev
+
+permissions:
+  contents: write
+  pull-requests: write
+
+name: release-please
+
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v4
+        with:
+          # this assumes that you have created a personal access token (PAT)
+          token: ${{ secrets.GITHUB_RELEASE_PLEASE_TOKEN }}
+          # optional. customize path to release-please-config.json
+          config-file: release-please-config.json
+          # optional. customize path to .release-please-manifest.json
+          manifest-file: .release-please-manifest.json
+
+      # The logic below handles the npm publication:
+      - uses: actions/checkout@v4
+        # these if statements ensure that a publication only occurs when
+        # a new release is created:
+        if: ${{ steps.release.outputs.release_created }}
+      - name: Install pnpm
+        uses: pnpm/action-setup@v3
+        with:
+          version: '^8.15'
+        if: ${{ steps.release.outputs.release_created }}
```

Comment:
I reckon good Github action hygiene is to separate the actual publish into a separate job. Then, we get more detailed action reports in PRs. 

One job can pass outputs to another in the same way you can from a step. Examples in extension of this.